---
 clang/lib/Driver/ToolChains.cpp |    2 ++
 1 file changed, 2 insertions(+)

Index: llvm-toolchain-5.0-5.0~+rc2/clang/lib/Driver/ToolChains/Gnu.cpp
===================================================================
--- llvm-toolchain-5.0-5.0~+rc2.orig/clang/lib/Driver/ToolChains/Gnu.cpp
+++ llvm-toolchain-5.0-5.0~+rc2/clang/lib/Driver/ToolChains/Gnu.cpp
@@ -15,6 +15,7 @@
 #include "Arch/SystemZ.h"
 #include "CommonArgs.h"
 #include "clang/Basic/VirtualFileSystem.h"
+#include "clang/Basic/Version.h"
 #include "clang/Config/config.h" // for GCC_INSTALL_PREFIX
 #include "clang/Driver/Compilation.h"
 #include "clang/Driver/Driver.h"
@@ -2368,6 +2369,7 @@ void Generic_GCC::AddClangCXXStdlibInclu
     addLibStdCxxIncludePaths(DriverArgs, CC1Args);
     break;
   }
+  addSystemInclude(DriverArgs, CC1Args, "/usr/include/clang/" + std::string(CLANG_VERSION_STRING) + "/include/");
 }
 
 std::string Generic_GCC::findLibCxxIncludePath() const {
