Index: llvm-toolchain-5.0-5.0.1/clang/lib/Driver/ToolChains/CommonArgs.cpp
===================================================================
--- llvm-toolchain-5.0-5.0.1.orig/clang/lib/Driver/ToolChains/CommonArgs.cpp
+++ llvm-toolchain-5.0-5.0.1/clang/lib/Driver/ToolChains/CommonArgs.cpp
@@ -600,9 +600,8 @@ collectSanitizerRuntimes(const ToolChain
 static void addLibFuzzerRuntime(const ToolChain &TC,
                                 const ArgList &Args,
                                 ArgStringList &CmdArgs) {
-    StringRef ParentDir = llvm::sys::path::parent_path(TC.getDriver().InstalledDir);
-    SmallString<128> P(ParentDir);
-    llvm::sys::path::append(P, "lib", "libLLVMFuzzer.a");
+    SmallString<128> P;
+    llvm::sys::path::append(P, TC.getDriver().Dir, "/../lib" CLANG_LIBDIR_SUFFIX, "libFuzzer.a");
     CmdArgs.push_back(Args.MakeArgString(P));
     TC.AddCXXStdlibLibArgs(Args, CmdArgs);
 }
